framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -311,8 +423,153 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
 		return false;
 	}
 
-	// eslint-disable-next-line @typescript-eslint/no-empty-function
-	private async _createCCU() {}
+	private async _calculateInboxUpdate(
+		sendingChainID: Buffer,
+		certificateStateRoot: Buffer,
+	): Promise<InboxUpdate> {
+		// TODO: Fetch as many CCMs as configured in _ccuFrequency.ccm, between lastCertificateHeight and certificateHeight - After #7569
+		const serialzedCCMs = this._chainConnectorState.crossChainMessages
+			.slice(0, this._ccuFrequency.ccm)
+			.map(ccm => codec.encode(ccmSchema, ccm));
+
+		const outboxKey = this._rawStateStoreKey(sendingChainID);
+
+		const stateProveResponse = await this._sidechainAPIClient.invoke<ProveResponse>('state_prove', {
+			stateRoot: certificateStateRoot,
```

Comment:
`stateRoot` is not required in the request, it's only possible to generate the proof against the current state